File: src/Adapter/src/Serie.java
import java.util.ArrayList;

public class Serie extends Midia {
    private int numeroTemporadas;
    private ArrayList<String> criadores;
    private ArrayList<Temporada> temporadas;

    // Getters and Setters

    public int getNumeroTemporadas() {
        return numeroTemporadas;
    }

    public void setNumeroTemporadas(int numeroTemporadas) {
        this.numeroTemporadas = numeroTemporadas;
    }

    public ArrayList<String> getCriadores() {
        return criadores;
    }

    public void setCriadores(ArrayList<String> criadores) {
        this.criadores = criadores;
    }

    public ArrayList<Temporada> getTemporadas() {
        return temporadas;
    }

    public void setTemporadas(ArrayList<Temporada> temporadas) {
        this.temporadas = temporadas;
    }

    public ArrayList<Temporada> listarTemporada() {
        return temporadas;
    }
}

File: src/Adapter/src/models/serie.ts
import Midia from './midia';
import Temporada from './temporada';

class Serie extends Midia {
    private numeroTemporadas: number = 0;
    private criadores: string[] = [];
    private temporadas: Temporada[] = [];

    // Getters and Setters
    public getNumeroTemporadas(): number {
        return this.numeroTemporadas;
    }

    public setNumeroTemporadas(numeroTemporadas: number): void {
        this.numeroTemporadas = numeroTemporadas;
    }

    public getCriadores(): string[] {
        return this.criadores;
    }

    public setCriadores(criadores: string[]): void {
        this.criadores = criadores;
    }

    public getTemporadas(): Temporada[] {
        return this.temporadas;
    }

    public setTemporadas(temporadas: Temporada[]): void {
        this.temporadas = temporadas;
    }

    public listarTemporada(): Temporada[] {
        return this.temporadas;
    }
}

export default Serie;